src/module/gate.rs

```diff
@@ -0,0 +1,50 @@
+use mixlab_protocol::GateState;
+
+use crate::engine::{Sample, CHANNELS};
+use crate::module::Module;
+
+#[derive(Debug)]
+pub struct Gate {
+    params: GateState,
+}
+
+impl Module for Gate {
+    type Params = GateState;
+    type Indication = ();
+
+    fn create(params: Self::Params) -> (Self, Self::Indication) {
+        (Gate { params }, ())
+    }
+
+    fn params(&self) -> Self::Params {
+        self.params.clone()
+    }
+
+    fn update(&mut self, new_params: Self::Params) -> Option<Self::Indication> {
+        self.params = new_params;
+        None
+    }
+
+    fn run_tick(&mut self, _t: u64, _inputs: &[Option<&[Sample]>], outputs: &mut [&mut [Sample]]) -> Option<Self::Indication> {
+        let len = outputs[0].len() / CHANNELS;
+
+        let value = match self.params {
+            GateState::Open => 1.0,
+            GateState::Closed => 0.0,
+        };
+
+        for i in 0..len {
+            outputs[0][i] = value;
```

Comment:
This will only output half a tick's worth of samples because of the divide by `CHANNELS` above. Diving by `CHANNELS` makes sense when we write two samples in the loop but not when we just want to fill the buffer

Reply:
ah sure, will fix it up. bad copy and paste